src/main/java/com/exasol/spark/common/Option.java

```diff
@@ -0,0 +1,99 @@
+package com.exasol.spark.common;
+
+/**
+ * Each {@link Option} is a supported entry for {@link ExasolOptions}.
+ *
+ * Using {@link Option#key()} you can check if the option is set and retrieve its value.
+ */
+public enum Option {
+
+    /** Name for Exasol table. */
+    TABLE("TABLE", "Parameter name for Exasol table."), //
+
+    /** Name for Exasol query. */
+    QUERY("QUERY", "Parameter name for Exasol query."), //
+
+    /** Exasol database connection JDBC URL. */
+    JDBC_URL("JDBC_URL", "Parameter name for Exasol database connection JDBC URL."), //
+
+    /** Exasol database username. */
+    USERNAME("USERNAME", "Parameter name for Exasol database username."), //
+
+    /** Exasol database password. */
+    PASSWORD("PASSWORD", "Parameter name for Exasol database password."), //
+
+    /** Number of Spark job partitions. */
+    NUMBER_OF_PARTITIONS("numPartitions", "Parameter name for setting number of Spark job partitions."), //
+
+    /** Default number of Spark job partitions. */
+    DEFAULT_NUMBER_OF_PARTITIONS("8", "Default number for Spark job partitions."), //
+
+    /** Maximum allowed number of Spark job partitions. */
+    MAX_ALLOWED_NUMBER_OF_PARTITIONS("1000", "Maximum allowed number for Spark job partitions."), //
+
+    /** Boolean parameter name to indicate local and CI environment. */
+    REPLACE_LOCALHOST_BY_DEFAULT_S3_ENDPOINT("replaceLocalhostByDefaultS3Endpoint",
+            "Boolean parameter name to indicate local and CI environment."), //
+
+    /**
+     * Parameter name for intermediate data location for writing.
+     *
+     * It is internal parameter, not required to be set by users.
+     */
+    INTERMEDIATE_DATA_PATH("PATH", "Parameter name for location for writing, to store intermediate data."), //
+
+    /**
+     * Parameter name for S3 bucket key inside intermediate data path.
+     *
+     * It is internal parameter, not required to be set by users.
+     */
+    WRITE_S3_BUCKET_KEY("writeS3BucketKey",
+            "Parameter name for bucket key that will be created in intermediate data path."), //
```

Comment:
If you want you could also add an (optional, boolean) argument `isPublic` to the constructor.
Or (more fancy) another enum, e.g. `enum Usage { PUBLIC, INTERNAL }`.
By this the options can be made introspective, i.e. each instance knows whether it is public or internal.
A CLI for example can later make use of this and report only public options.

Reply:
That is good idea. For now we could keep it like this. 

Here, I wanted to describe to us the developer that these one will be set in the code (will be overwritten anyway if users set these). Later in documentation we will mention only the ones required by users.